uportal-war/src/main/java/org/apereo/portal/security/provider/cas/CasAssertionSecurityContext.java

```diff
@@ -167,6 +200,23 @@ protected void copyAssertionAttributesToUserAttributes(Assertion assertion) {
             log.debug("Adding attribute '{}' from Assertion with value '{}'; runtime type of value is {}",
                     attrEntry.getKey(), attrEntry.getValue(), attrEntry.getValue().getClass().getName());
 
+            // Check for credential
+            if (decryptCredentialToPassword && cipher != null && attrEntry.getKey().equals(CREDENTIAL_KEY)) {
+                try {
+                    final String encPwd = (String) (attrEntry.getValue() instanceof List
+                            ? ((List) attrEntry.getValue()).get(0) : attrEntry.getValue());
+                    byte[] cred64 = Base64.getDecoder().decode(encPwd);
```

Comment:
This class was introduced in Java 8. To backport this to 4.3.x, I suppose it needs to work with Java 7...